aten/src/ATen/TensorIterator.cpp

```diff
@@ -1281,12 +1302,25 @@ void TensorIteratorBase::build(TensorIteratorConfig& config) {
   view_offsets_ = DimVector(ndim_offsets, 0);
 }
 
-void TensorIterator::set_output(int64_t output_idx, IntArrayRef sizes, IntArrayRef strides, TensorOptions options, DimnameList names) {
+void TensorIterator::set_output(
+    int64_t output_idx, IntArrayRef sizes, IntArrayRef strides,
+    TensorOptions options, DimnameList names,
+    c10::optional<TensorQuantizationOptions> quant_options) {
   auto& op = operands_[output_idx];
   TORCH_INTERNAL_ASSERT_DEBUG_ONLY(output_idx < num_outputs_);
   if (!op.tensor.defined()) {
       if (strides.empty()) {
-          op.tensor = at::empty(sizes, options);
+          if (options.dtype() == at::kQInt8 || options.dtype() == at::kQUInt8 ||
+              options.dtype() == at::kQInt32) {
+            // quantized path
+            TORCH_INTERNAL_ASSERT(quant_options.has_value());
+            op.tensor = at::_empty_affine_quantized(
+                sizes, options, (*quant_options).q_scale,
+                (*quant_options).q_zero_point);
+          } else {
+            // non-quantized path
+            op.tensor = at::empty(sizes, options);
+          }
       } else {
           op.tensor = at::empty_strided(sizes, strides, options);
       }
```

Comment:
Below, there is logic for handling what happens in the `out=` case. Imagine that you do indexing into a pre-existing quantized tensor. What should the quantization parameters be? What if the tensor is to be *resized* in this situation?

Reply:
asserted it out for now